pkg/api/customization/vsphere/handler.go

```diff
@@ -0,0 +1,187 @@
+}
+
+var dataFields = map[string]string{
+	"username": "vmwarevspherecredentialConfig-username",
+	"password": "vmwarevspherecredentialConfig-password",
+	"host":     "vmwarevspherecredentialConfig-vcenter",
+	"port":     "vmwarevspherecredentialConfig-vcenterPort",
+}
+
+type handler struct {
+	schemas       *types.Schemas
+	secretsLister v1.SecretLister
+}
+
+func NewVsphereHandler(scaledContext *config.ScaledContext) http.Handler {
+	return &handler{
+		schemas:       scaledContext.Schemas,
+		secretsLister: scaledContext.Core.Secrets(namespace.GlobalNamespace).Controller().Lister(),
+	}
+}
+
+func (v *handler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	res.Header().Set("Content-Type", "application/json")
+	fieldName := mux.Vars(req)["field"]
+	dc := req.FormValue("dataCenter")
+
+	if fieldName == "" || !validFieldName(fieldName) {
+		util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: invalid field name", httperror.NotFound.Code))
+		return
+	}
+
+	cc, err := v.getCloudCredential(req)
+	if err != nil {
+		util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: %s", httperror.InvalidBodyContent.Code, err.Error()))
+		return
+	}
+
+	var data []string
+	if fieldNames[fieldName] == SoapFinder {
+		data, err = processSoapFinder(req.Context(), fieldName, cc, dc)
+		if err != nil {
+			util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: %s", httperror.InvalidBodyContent.Code, err.Error()))
+			return
+		}
+	}
+
+	if fieldNames[fieldName] == ContentLibraryManager {
+		l := req.FormValue("library")
+		data, err = processContentLibraryManager(req.Context(), fieldName, cc, l)
+		if err != nil {
+			util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: %s", httperror.InvalidBodyContent.Code, err.Error()))
+			return
+		}
+	}
+
+	if fieldNames[fieldName] == TagsManager {
+		c := req.FormValue("category")
+		data, err = processTagsManager(req.Context(), fieldName, cc, c)
+		if err != nil {
+			util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: %s", httperror.InvalidBodyContent.Code, err.Error()))
+			return
+		}
+	}
+
+	if fieldNames[fieldName] == CustomFieldsFinder {
+		data, err = processCustomFieldsFinder(req.Context(), cc)
+		if err != nil {
+			util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: %s", httperror.InvalidBodyContent.Code, err.Error()))
+			return
+		}
+	}
+
+	js, err := json.Marshal(map[string][]string{
+		"data": data,
+	})
+
+	if err != nil {
+		util.ReturnHTTPError(res, req, httperror.NotFound.Status, fmt.Sprintf("%s: %s", httperror.ServerError.Code, err.Error()))
+		return
+	}
+
+	res.WriteHeader(http.StatusOK)
+	res.Write(js)
+}
+
+func (v *handler) getCloudCredential(req *http.Request) (*corev1.Secret, error) {
+	credID := req.FormValue("cloudCredentialId")
+	if credID == "" {
+		return nil, fmt.Errorf("cloud credential required")
+	}
+
+	var accessCred client.CloudCredential
```

Comment:
Can you add a comment above this to clarify why we don't need it? It holds no info, the secret holds our world etc.